test/functional/api/cas/casadm.py

```diff
@@ -59,42 +60,88 @@ def start_cache(
             shortcut=shortcut,
         )
     )
+
     if output.exit_code != 0:
         raise CmdException("Failed to start cache.", output)
-    return Cache(cache_dev)
+
+    if not _cache_id:
+        from api.cas.casadm_parser import get_caches
+
+        cache_list = get_caches()
+        # compare path of old and new caches, returning the only one created now.
+        # This will be needed in case cache_id not present in cli command
+
+        new_cache = next(
+            iter([cache for cache in cache_list if cache.cache_device.path == cache_dev.path])
+        )
+        _cache_id = new_cache.cache_id
+
+    cache = Cache(cache_id=int(_cache_id), device=cache_dev, cache_line_size=_cache_line_size)
+    TestRun.dut.cache_list.append(cache)
```

Comment:
I don't like the concept of keeping TestRun.dut.cache_list/core_list, but I really really don't like that casadm.py is in charge of items on those lists. I feel like it's beyond casadm's responsibilities. Just saying :(